src/commands/kv/bucket.rs

```diff
@@ -0,0 +1,103 @@
+extern crate base64;
+
+use crate::commands::kv::delete_bulk::delete_bulk;
+use crate::commands::kv::write_bulk::write_bulk;
+
+use cloudflare::endpoints::workerskv::write_bulk::KeyValuePair;
+
+use walkdir::WalkDir;
+
+use std::ffi::OsString;
+use std::fs::metadata;
+use std::path::Path;
+
+use failure::bail;
+
+use crate::terminal::message;
+
+pub fn upload(namespace_id: &str, filename: &Path) -> Result<(), failure::Error> {
+    let pairs: Result<Vec<KeyValuePair>, failure::Error> = match metadata(filename) {
+        Ok(ref file_type) if file_type.is_dir() => parse_directory(filename),
+        Ok(_file_type) => {
+            // any other file types (namely, symlinks)
+            bail!("wrangler kv:bucket upload takes a directory",)
+        }
+        Err(e) => bail!(e),
+    };
+
+    write_bulk(namespace_id, pairs?)
+}
+
+pub fn delete(namespace_id: &str, filename: &Path) -> Result<(), failure::Error> {
+    let keys: Result<Vec<String>, failure::Error> = match metadata(filename) {
+        Ok(ref file_type) if file_type.is_dir() => {
+            let key_value_pairs = parse_directory(filename)?;
+            Ok(key_value_pairs
+                .iter()
+                .map(|kv| kv.key.clone())
+                .collect::<Vec<_>>())
+        }
+        Ok(_file_type) => {
+            // any other file types (namely, symlinks)
+            bail!(
+                "{} should be a file or directory, but is a symlink",
+                filename.display()
+            )
+        }
+        Err(e) => bail!(e),
+    };
+
+    delete_bulk(namespace_id, keys?)
+}
+
+fn parse_directory(directory: &Path) -> Result<Vec<KeyValuePair>, failure::Error> {
+    let mut upload_vec: Vec<KeyValuePair> = Vec::new();
+    for entry in WalkDir::new(directory) {
+        let entry = entry.unwrap();
```

Comment:
Per Steve's comments, `let entry = entry.unwrap().path();`

Reply:
tried this; the compiler complained about a dropped borrow and suggested re-adding the let...

Reply:
*shakes fist* BORROW CHECKER!!!